linker/shared/src/main/scala/org/scalajs/linker/backend/emitter/CoreJSLib.scala

```diff
@@ -725,6 +725,37 @@ private[emitter] object CoreJSLib {
     private def defineDispatchFunctions(): Tree = {
       val instance = varRef("instance")
 
+      val methodsInRepresentativeClasses = globalKnowledge
+        .methodsInRepresentativeClasses()
+        .filter(_._2 != toStringMethodName) // toString() is special
+        .groupBy(_._2)
+        .map(kv => (kv._1, kv._2.map(_._1)))
+        .toList
+        .sortBy(_._1.nameString) // sort for determinism
+
+      /* Hijacked classes ordered by priority for the dispatch, except "small"
+       * number types (which are covered by Double).
+       * Those with `typeof`-based tests come first.
+       */
+      val nonSmallNumberHijackedClasses = List(
+        BoxedDoubleClass,
+        BoxedStringClass,
+        BoxedBooleanClass,
+        BoxedUnitClass,
+        BoxedCharacterClass
+      ) ::: (
+        if (allowBigIntsForLongs) List(BoxedLongClass) else Nil
+      ) ::: List(
+        BoxedCharacterClass
+      )
+
+      val allOrderedHijackedClasses = List(
```

Comment:
This list already exists as `ClassEmitter#RuntimeSubclassesOfDouble`. In fact there is very similar code in `genSingleInstanceTests`. Consolidate? (Including removal of `Long` and small numbers?).

Reply:
Indeed. I've consolidated all of that.